### PR TITLE
feat: add possibility to pass custom button label in collapsible fieldset widget

### PR DESCRIPTION
**Pull request by VolodymyrKovalM**
**What is the problem this PR is trying to solve?**
https://jira.talendforge.org/browse/TMC-14619

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

### CODE REVIEW

File: packages/forms/src/UIForm/fieldsets/Array/DefaultArrayTemplate.component.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ function DefaultArrayTemplate(props) {
 				className={classNames(theme['tf-array-add'], 'tf-array-add')}
 				bsStyle={'info'}
 				onClick={onAdd}
-				label={t('ARRAY_ADD_ELEMENT', { defaultValue: 'New Element' })}
+				label={schema.btnLabel || t('ARRAY_ADD_ELEMENT', { defaultValue: 'New Element' })}
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/fieldsets/Array/DefaultArrayTemplate.component.js]: To pass custom properties, don't put them directly in uiSchema. There is a sort of convention that we use in forms package, add an option object, where you can add the properties you want.

Please add the entry in documentation (readme in array folder), and a storybook.

File: output/forms.eslint.txt [hunk 2]
```diff
@@ -9,9 +9,12 @@ The react/require-extension rule is deprecated. Please use the import/extensions
   10:11  error  Trailing spaces not allowed                     no-trailing-spaces
   12:1   error  Line 12 exceeds the maximum line length of 100  max-len
 
+/home/travis/build/Talend/ui/packages/forms/src/UIForm/fieldsets/Array/DefaultArrayTemplate.component.js
+  26:3  error  'options' is missing in props validation  react/prop-types
```
Comment (jsomsanith) [inline output/forms.eslint.txt]: Please fix the linter error

File: packages/forms/src/UIForm/fieldsets/Array/DefaultArrayTemplate.component.js [hunk 3]
```diff
@@ -34,7 +35,7 @@ function DefaultArrayTemplate(props) {
 				className={classNames(theme['tf-array-add'], 'tf-array-add')}
 				bsStyle={'info'}
 				onClick={onAdd}
-				label={t('ARRAY_ADD_ELEMENT', { defaultValue: 'New Element' })}
+				label={options.btnLabel || t('ARRAY_ADD_ELEMENT', { defaultValue: 'New Element' })}
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fieldsets/Array/DefaultArrayTemplate.component.js]: this is the only options you wanna pass, so why an object, why not just addBtnLabel as a string directly in the uiSchema ?

Comment (VolodymyrKovalM) [reply packages/forms/src/UIForm/fieldsets/Array/DefaultArrayTemplate.component.js]: @jmfrancois so, initially I've done like this, but according to @jsomsanith comments there is a convention to not put custom properties directly in uiSchema

please read @jsomsanith comments above

File: packages/forms/src/UIForm/fieldsets/Array/DefaultArrayTemplate.component.js [hunk 4]
```diff
@@ -74,6 +75,7 @@ if (process.env.NODE_ENV !== 'production') {
 		renderItem: PropTypes.func.isRequired,
 		schema: PropTypes.object.isRequired,
 		value: PropTypes.arrayOf(PropTypes.object).isRequired,
+		options: PropTypes.object,
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fieldsets/Array/DefaultArrayTemplate.component.js]: please use shape here with the btnLabel string inside

File: packages/forms/src/UIForm/fieldsets/Array/README.md [hunk 5]
```diff
@@ -179,3 +180,4 @@ Custom Array Template props
 | renderItem | `function` | `renderItem(index)` function to call to render an element. |
 | schema | `object` | The merged schema. |
 | value | `any` | The array value. |
+| options | `object` | If you want to pass some custom props you can pass them in this object and use inside the component
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fieldsets/Array/README.md]: please list all supported options

**Pull request merged**